tt-media-server/open_ai_api/tts.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
+
+import json
+from typing import Optional
+
+from config.constants import AudioResponseFormat
+from domain.text_to_speech_request import TextToSpeechRequest
+from fastapi import (
+    APIRouter,
+    Depends,
+    Form,
+    HTTPException,
+    Request,
+    Response,
+    Security,
+)
+from fastapi.responses import StreamingResponse
+from model_services.base_service import BaseService
+from resolver.service_resolver import service_resolver
+from security.api_key_cheker import get_api_key
+
+
+async def parse_tts_request(
+    request: Request,
+    text: Optional[str] = Form(None),
+    stream: Optional[bool] = Form(False),
+    response_format: Optional[str] = Form(AudioResponseFormat.VERBOSE_JSON.value),
+    speaker_id: Optional[str] = Form(None),
+    speaker_embedding: Optional[str] = Form(None),
+) -> TextToSpeechRequest:
+    """Parse TTS request from form data or JSON"""
+    content_type = request.headers.get("content-type", "").lower()
+
+    if text is not None or content_type == "multipart/form-data":
+        # Form data request
+        return TextToSpeechRequest(
+            text=text,
+            stream=stream or False,
+            speaker_id=speaker_id,
+            speaker_embedding=speaker_embedding,
+        )
+
+    if "application/json" in content_type:
+        # JSON request
+        json_body = await request.json()
+        return TextToSpeechRequest(**json_body)
+
+    raise HTTPException(
+        status_code=400,
+        detail="Use either multipart/form-data with text parameter or application/json",
+    )
+
+
+tts_router = APIRouter()
+
+
+@tts_router.post("/tts")
+async def text_to_speech(
+    tts_request: TextToSpeechRequest = Depends(parse_tts_request),
+    service = Depends(service_resolver),
+    api_key: str = Security(get_api_key),
+):
+    """
+    Convert text to speech using the provided request.
+    Supports both streaming and non-streaming based on the 'stream' field in the request.
+
+    Returns:
+        The audio response or StreamingResponse based on request.stream field.
+
+    Raises:
+        HTTPException: If text-to-speech fails.
+    """
+    return await handle_tts_request(tts_request, service)
+
+
+async def handle_tts_request(tts_request, service):
+    try:
+        if not tts_request.stream:
+            result = await service.process_request(tts_request)
+            if tts_request.response_format.lower() == AudioResponseFormat.TEXT.value:
+                return Response(content=result.audio, media_type="audio/wav")
+            return get_dict_response(result)
+        else:
+            try:
+                service.scheduler.check_is_model_ready()
+            except Exception:
+                raise HTTPException(status_code=405, detail="Model is not ready")
```

Comment:
We could utilize here guard clause, like this:
```suggestion
        if not tts_request.stream:
            result = await service.process_request(tts_request)
            if tts_request.response_format.lower() == AudioResponseFormat.TEXT.value:
                return Response(content=result.audio, media_type="audio/wav")
            return get_dict_response(result)
        
        # Streaming path
        try:
            service.scheduler.check_is_model_ready()
        except Exception:
            raise HTTPException(status_code=405, detail="Model is not ready")
```